lib/src/services/sources_api.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:convert';
+
+import 'package:omise_dart/src/enums/omise_api_errors.dart';
+import 'package:omise_dart/src/exceptions/omise_api_exception.dart';
+import 'package:omise_dart/src/models/requests/create_source_request.dart';
+import 'package:omise_dart/src/models/responses/omise_api_error.dart';
+import 'package:omise_dart/src/models/responses/source.dart';
+import 'package:omise_dart/src/omise_http_client.dart';
+
+/// Provides methods for interacting with the source endpoints of the Omise API.
+class SourcesApi {
+  /// Creates an instance of [SourcesApi] with the provided [OmiseHttpClient].
+  ///
+  /// The [OmiseHttpClient] is used to make HTTP requests to the Omise API.
+  SourcesApi(this.httpClient);
+
+  final OmiseHttpClient httpClient;
+
+  /// Creates a new source using the provided [createSourceRequest].
+  ///
+  /// Sends a POST request to the '/sources' endpoint with the source creation request data.
+  ///
+  /// Returns a [Source] if the request is successful (HTTP status code 200).
+  ///
+  /// Throws an [OmiseApiException] if the request fails. The exception includes
+  /// the error message and status code, along with the response from the API.
+  Future<Source> create(CreateSourceRequest createSourceRequest) async {
+    final response = await httpClient.post(
+      '/sources',
+      body: createSourceRequest.toJson(),
+    );
+    final decodedBody = json.decode(response.body);
+    if (response.statusCode == 200) {
+      return Source.fromJson(
+        decodedBody,
+      );
+    } else {
```

Comment:
Do you think this block would still be readable without `else`?

Reply:
Yeah can work, will change.